File: src/components/AiPredict.js
/**
 * MyKing AI Prediction Component
 * 80% 실력분석 / 10% 운세 / 10% 날씨
 */
export function renderAiPredict(match, container) {
  const { home_team, away_team, id } = match;
  const wrapper = document.createElement('div');
  wrapper.className = 'ai-predict-wrapper';

  wrapper.innerHTML = `
    <button class="ai-predict-btn" data-match-id="${id}">
      <span class="ai-btn-icon">🔮</span>
      <span class="ai-btn-text">MyKing AI 예측</span>
      <span class="ai-btn-badge">BETA</span>
    </button>
    <div class="ai-panel hidden" data-match-id="${id}">
      <div class="ai-loading">
        <div class="ai-spinner"></div>
        <span>MyKing이 데이터를 분석중입니다...</span>
      </div>
    </div>
  `;

  // Attach click handler
  const btn = wrapper.querySelector('.ai-predict-btn');
  const panel = wrapper.querySelector('.ai-panel');

  btn.addEventListener('click', async () => {
    panel.classList.toggle('hidden');
    if (!panel.classList.contains('hidden') && !panel.dataset.loaded) {
      panel.dataset.loaded = 'true';
      await loadAiPrediction(id, panel);
    }
  });

  container.appendChild(wrapper);
}

async function loadAiPrediction(matchId, panel) {
  try {
    // Try backend API first
    const res = await fetch(`/api/predict/ai/${matchId}`);
    if (res.ok) {
      const data = await res.json();
      renderPredictionResult(data, panel);
      return;
    }
  } catch (e) {
    // Fallback to mock
  }

  // Fallback: generate mock AI prediction
  const mock = await generateMockPrediction(matchId);
  renderPredictionResult(mock, panel);
}

async function generateMockPrediction(matchId) {
  // Generate a deterministic-ish mock based on matchId
  const seed = matchId.charCodeAt(1) || 42;

  const teams = {
    m1: { home: '브라질', away: '독일', homeRank: 1, awayRank: 10 },
    m2: { home: '대한민국', away: '일본', homeRank: 23, awayRank: 15 },
    m3: { home: '아르헨티나', away: '사우디아라비아', homeRank: 2, awayRank: 56 },
    m4: { home: '프랑스', away: '포르투갈', homeRank: 3, awayRank: 6 },
    m5: { home: '스페인', away: '네덜란드', homeRank: 5, awayRank: 8 },
    m6: { home: '브라질', away: '아르헨티나', homeRank: 1, awayRank: 2 },
  };

  const t = teams[matchId] || { home: '팀A', away: '팀B', homeRank: 15, awayRank: 15 };
  const rankDiff = t.awayRank - t.homeRank;
  const homePower = Math.min(95, Math.max(40, 50 + rankDiff * 1.5 + (seed % 10)));
  const awayPower = Math.min(95, Math.max(40, 50 - rankDiff * 1.5 + ((seed * 3) % 10)));

  const total = homePower + awayPower;
  const homeSkill = Math.round((homePower / total) * 100);
  const awaySkill = 100 - homeSkill;

  const homeLuck = ((seed * 7 + 13) % 21) - 8;
  const isHomeLucky = homeLuck > 0;

  const weathers = ['맑음 ☀️', '흐림 ⛅', '비 🌧️', '폭염 🔥', '강풍 💨'];
  const weather = weathers[seed % 5];

  const homeTotal = homeSkill * 0.8 + (homeLuck + 50) * 0.1 + 10 * 0.1;
  const awayTotal = awaySkill * 0.8 + (-homeLuck + 50) * 0.1 + 9 * 0.1;
  const drawBase = 20;
  const sum = homeTotal + awayTotal + drawBase;

  const homePct = Math.round((homeTotal / sum) * 100);
  const awayPct = Math.round((awayTotal / sum) * 100);
  const drawPct = 100 - homePct - awayPct;

  const hg = Math.max(0, Math.round((homePower / 100) * (2 + (seed % 3))));
  const ag = Math.max(0, Math.round((awayPower / 100) * (1 + ((seed * 2) % 2))));

  const forms = ['WWWDW', 'WDLWW', 'WWLWD', 'LDWLL', 'WDWWL'];
  const homeForm = forms[(seed * 3) % 5];
  const awayForm = forms[(seed * 7) % 5];

  return {
    match_id: matchId,
    home_team: t.home,
    away_team: t.away,
    predicted_score: { home: hg > ag ? hg : hg + 1, away: hg > ag ? ag : ag + 1 },
    winner: hg > ag ? t.home : t.away,
    probabilities: { home_win: homePct, draw: drawPct, away_win: awayPct },
    analysis: {
      skill_80: {
        home_power: Math.round(homePower),
        away_power: Math.round(awayPower),
        home_attack: `⚔️ ${(2 + seed % 5 / 5).toFixed(1)} goals/match`,
        away_attack: `⚔️ ${(1.5 + (seed * 3) % 5 / 5).toFixed(1)} goals/match`,
        home_defense: `🛡️ ${(0.5 + (seed % 10) / 10).toFixed(1)} conceded/match`,
        away_defense: `🛡️ ${(0.8 + (seed * 7) % 10 / 10).toFixed(1)} conceded/match`,
        home_form: `최근 ${homeForm}`,
        away_form: `최근 ${awayForm}`,
        key_player: getKeyPlayer(t.home) + ' vs ' + getKeyPlayer(t.away),
        rank_diff: `FIFA 랭킹: ${t.homeRank}위 vs ${t.awayRank}위`,
        home_pct: homeSkill,
        away_pct: awaySkill,
      },
      luck_10: {
        message: isHomeLucky
          ? ['별자리가 상승세입니다! 🌟', '오늘은 🐯 호랑이 기운이 가득합니다!', '황금 물결이 당신을 감쌉니다 💛'][seed % 3]
          : ['수성 역행일지도... 🌑', '검은 고양이가 길을 건넜습니다 🐈‍⬛', '살짝 불운이 깃든 날입니다 💫'][seed % 3],
        value: Math.abs(homeLuck) * 2,
        is_lucky: isHomeLucky,
      },
      weather_10: {
        condition: weather,
        temperature: ['28°C', '22°C', '18°C', '35°C', '20°C'][seed % 5],
        home_advantage: 1.0 + (seed % 3) * 0.05,
        away_disadvantage: 0.9 + (seed % 4) * 0.03,
      },
    },
    verdict: homePower - awayPower > 15
      ? `🏆 ${t.home}의 압도적 우세!`
      : homePower - awayPower > 5
      ? `⭐ ${t.home}이(가) 다소 유리합니다.`
      : Math.abs(homePower - awayPower) <= 5
      ? '⚖️ 초접전이 예상됩니다!'
      : `💪 ${t.away}이(가) 객관적 전력에서 앞서지만, 변수가 많습니다.`,
    confidence: '매우 높음',
  };
}

function getKeyPlayer(team) {
  const map = {
    '브라질': '비니시우스', '독일': '무시알라', '대한민국': '손흥민',
    '일본': '미토마', '아르헨티나': '메시', '사우디': '알다우사리',
    '사우디아라비아': '알다우사리', '프랑스': '음바페', '포르투갈': '호날두',
    '스페인': '야말', '네덜란드': '데파이',
  };
  return map[team] || 'N/A';
}

function renderPredictionResult(data, panel) {
  const { predicted_score, probabilities, analysis, verdict, confidence } = data;
  const s = analysis.skill_80;
  const l = analysis.luck_10;
  const w = analysis.weather_10;

  panel.innerHTML = `
    <div class="ai-header">
      <span class="ai-title">🔮 MyKing AI 분석</span>
      <span class="ai-confidence">신뢰도: ${confidence}</span>
    </div>

    <div class="ai-score-box">
      <div class="ai-team">
        <span class="ai-flag">${getFlag(data.home_team)}</span>
        <span class="ai-team-name">${data.home_team}</span>
      </div>
      <div class="ai-score">
        <span class="ai-score-num home">${predicted_score.home}</span>
        <span class="ai-score-colon">:</span>
        <span class="ai-score-num away">${predicted_score.away}</span>
      </div>
      <div class="ai-team">
        <span class="ai-team-name">${data.away_team}</span>
        <span class="ai-flag">${getFlag(data.away_team)}</span>
      </div>
    </div>

    <div class="ai-prob-bars">
      <div class="ai-prob-row">
        <span class="ai-prob-label home">${data.home_team}</span>
        <div class="ai-prob-bar-wrap">
          <div class="ai-prob-bar" style="width:${probabilities.home_win}%;background:var(--blue)"></div>
        </div>
        <span class="ai-prob-val">${probabilities.home_win}%</span>
      </div>
      <div class="ai-prob-row">
        <span class="ai-prob-label draw">무승부</span>
        <div class="ai-prob-bar-wrap">
          <div class="ai-prob-bar draw-bar" style="width:${probabilities.draw}%"></div>
        </div>
        <span class="ai-prob-val">${probabilities.draw}%</span>
      </div>
      <div class="ai-prob-row">
        <span class="ai-prob-label away">${data.away_team}</span>
        <div class="ai-prob-bar-wrap">
          <div class="ai-prob-bar" style="width:${probabilities.away_win}%;background:var(--purple)"></div>
        </div>
        <span class="ai-prob-val">${probabilities.away_win}%</span>
      </div>
    </div>

    <div class="ai-verdict">${verdict}</div>

    <div class="ai-breakdown">
      <div class="ai-bd-title">📊 분석 breakdown</div>

      <!-- 80% Skill -->
      <div class="ai-bd-section">
        <div class="ai-bd-header">
          <span class="ai-bd-icon">💪</span>
          <span class="ai-bd-label">실력 분석</span>
          <span class="ai-bd-weight">80%</span>
        </div>
        <div class="ai-bd-body">
          <div class="ai-compare-row">
            <span class="ai-compare-left">${s.home_power}</span>
            <span class="ai-compare-label">전력 지수</span>
            <span class="ai-compare-right">${s.away_power}</span>
          </div>
          <div class="ai-power-bar-wrap">
            <div class="ai-power-bar home" style="width:${s.home_pct}%"></div>
            <div class="ai-power-bar away" style="width:${s.away_pct}%"></div>
          </div>
          <div class="ai-detail">
            <div class="ai-detail-item"><span>${s.rank_diff}</span></div>
            <div class="ai-detail-item"><span>${s.key_player}</span></div>
            <div class="ai-detail-item"><span>${s.home_attack} / ${s.away_attack}</span></div>
            <div class="ai-detail-item"><span>${s.home_defense} / ${s.away_defense}</span></div>
            <div class="ai-detail-item"><span>${s.home_form} / ${s.away_form}</span></div>
          </div>
        </div>
      </div>

      <!-- 10% Luck -->
      <div class="ai-bd-section">
        <div class="ai-bd-header">
          <span class="ai-bd-icon">🍀</span>
          <span class="ai-bd-label">운세 분석</span>
          <span class="ai-bd-weight">10%</span>
        </div>
        <div class="ai-bd-body">
          <div class="ai-luck-msg ${l.is_lucky ? 'good' : 'bad'}">${l.message}</div>
          <div class="ai-luck-bar">
            <div class="ai-luck-fill ${l.is_lucky ? 'good' : 'bad'}" style="width:${l.value}%"></div>
          </div>
          <div class="ai-luck-label">행운 수치: ${l.value}%</div>
        </div>
      </div>

      <!-- 10% Weather -->
      <div class="ai-bd-section">
        <div class="ai-bd-header">
          <span class="ai-bd-icon">🌤️</span>
          <span class="ai-bd-label">날씨/컨디션</span>
          <span class="ai-bd-weight">10%</span>
        </div>
        <div class="ai-bd-body">
          <div class="ai-weather-display">
            <span>${w.condition}</span>
            <span>${w.temperature}</span>
          </div>
          <div class="ai-weather-effect">
            <span>🏠 홈 이점: +${Math.round((w.home_advantage - 1) * 100)}%</span>
            <span>✈️ 원정 페널티: -${Math.round((1 - w.away_disadvantage) * 100)}%</span>
          </div>
        </div>
      </div>
    </div>
  `;
}

function getFlag(team) {
  const map = {
    '브라질': '🇧🇷', '독일': '🇩🇪', '대한민국': '🇰🇷', '일본': '🇯🇵',
    '아르헨티나': '🇦🇷', '사우디아라비아': '🇸🇦', '프랑스': '🇫🇷',
    '포르투갈': '🇵🇹', '스페인': '🇪🇸', '네덜란드': '🇳🇱',
    '팀A': '🏴', '팀B': '🏴',
  };
  return map[team] || '🏴';
}

File: backend/routes/ai_predict.py
"""
MyKing AI Prediction Engine
- 80% 실력 분석 (FIFA rank, recent form, H2H, goals)
- 10% 운세 (random luck factor)
- 10% 날씨/컨디션 (match-day conditions)
"""
from fastapi import APIRouter
import random
import math

router = APIRouter()

# ── Extended team database ──
TEAM_DB = {
    "브라질": {
        "rank": 1, "flag": "🇧🇷", "color": "#009739", "form": "WWWDW",
        "gf_avg": 2.4, "ga_avg": 0.7, "strength": 94,
        "style": "공격적", "key_player": "비니시우스",
        "coach": "도리바우 주니오르",
    },
    "독일": {
        "rank": 10, "flag": "🇩🇪", "color": "#000000", "form": "WDWWL",
        "gf_avg": 2.1, "ga_avg": 0.9, "strength": 88,
        "style": "조직적", "key_player": "무시알라",
        "coach": "율리안 나겔스만",
    },
    "대한민국": {
        "rank": 23, "flag": "🇰🇷", "color": "#C60C30", "form": "WLWDW",
        "gf_avg": 1.8, "ga_avg": 1.2, "strength": 78,
        "style": "역습", "key_player": "손흥민",
        "coach": "홍명보",
    },
    "일본": {
        "rank": 15, "flag": "🇯🇵", "color": "#FFFFFF", "form": "WWLWD",
        "gf_avg": 2.0, "ga_avg": 1.0, "strength": 82,
        "style": "기술적", "key_player": "미토마",
        "coach": "모리야스 하지메",
    },
    "아르헨티나": {
        "rank": 2, "flag": "🇦🇷", "color": "#75AADB", "form": "WWWWW",
        "gf_avg": 2.5, "ga_avg": 0.5, "strength": 96,
        "style": "균형적", "key_player": "메시",
        "coach": "리오넬 스칼로니",
    },
    "사우디아라비아": {
        "rank": 56, "flag": "🇸🇦", "color": "#006C35", "form": "LDWLL",
        "gf_avg": 1.1, "ga_avg": 1.8, "strength": 64,
        "style": "수비적", "key_player": "알다우사리",
        "coach": "로베르토 만치니",
    },
    "프랑스": {
        "rank": 3, "flag": "🇫🇷", "color": "#002395", "form": "WWWDW",
        "gf_avg": 2.3, "ga_avg": 0.8, "strength": 92,
        "style": "역동적", "key_player": "음바페",
        "coach": "디디에 데샹",
    },
    "포르투갈": {
        "rank": 6, "flag": "🇵🇹", "color": "#006600", "form": "WWWLW",
        "gf_avg": 2.0, "ga_avg": 0.9, "strength": 87,
        "style": "기술적", "key_player": "호날두",
        "coach": "로베르토 마르티네즈",
    },
    "스페인": {
        "rank": 5, "flag": "🇪🇸", "color": "#C60B1E", "form": "WWWDW",
        "gf_avg": 2.2, "ga_avg": 0.7, "strength": 89,
        "style": "티키타카", "key_player": "야말",
        "coach": "루이스 데 라 푸엔테",
    },
    "네덜란드": {
        "rank": 8, "flag": "🇳🇱", "color": "#FF6600", "form": "WDLWW",
        "gf_avg": 1.9, "ga_avg": 1.0, "strength": 85,
        "style": "토탈풋볼", "key_player": "데파이",
        "coach": "로날드 쿠만",
    },
}

# ── Weather conditions ──
WEATHERS = [
    {"condition": "맑음 ☀️", "temp": "28°C", "effect_home": 1.0, "effect_away": 1.0},
    {"condition": "흐림 ⛅", "temp": "22°C", "effect_home": 1.0, "effect_away": 0.95},
    {"condition": "비 🌧️", "temp": "18°C", "effect_home": 1.05, "effect_away": 0.90},
    {"condition": "폭염 🔥", "temp": "35°C", "effect_home": 1.10, "effect_away": 0.85},
    {"condition": "강풍 💨", "temp": "20°C", "effect_home": 1.02, "effect_away": 0.92},
]

# ── Fortune messages ──
LUCK_GOOD = [
    "오늘은 🐯 호랑이 기운이 가득합니다!",
    "별자리가 상승세입니다! 🌟",
    "황금 물결이 당신을 감쌉니다 💛",
    "행운의 숫자가 7입니다 🔢",
]
LUCK_BAD = [
    "오늘은 수성 역행일지도... 🌑",
    "검은 고양이가 길을 건넜습니다 🐈‍⬛",
    "살짝 불운이 깃든 날입니다 💫",
    "조심스러운 접근이 필요합니다 🧘",
]


def get_team_data(name):
    """Get team data with fallback for unknown teams"""
    if name in TEAM_DB:
        return TEAM_DB[name]
    return {
        "rank": 30, "flag": "🏴", "color": "#888888", "form": "WDLWD",
        "gf_avg": 1.5, "ga_avg": 1.3, "strength": 75,
        "style": "균형적", "key_player": "N/A", "coach": "N/A",
    }


def calculate_power_score(team, opponent):
    """Calculate base power score (0-100)"""
    t = get_team_data(team)
    o = get_team_data(opponent)

    # FIFA rank difference (up to 30 points)
    rank_diff = max(0, (o["rank"] - t["rank"]) / 55 * 30)

    # Form score (W=1, D=0.5, L=0)
    form_score = sum(1 if c == "W" else 0.5 if c == "D" else 0 for c in t["form"]) / 5 * 25

    # Strength rating
    strength_score = t["strength"] / 100 * 25

    # Goals average
    goal_score = (t["gf_avg"] / 3) * 20

    total = rank_diff + form_score + strength_score + goal_score
    return min(99, max(1, total))


def calculate_prediction(match):
    """Calculate AI prediction for a match"""
    home = match.get("home_team", "")
    away = match.get("away_team", "")
    home_data = get_team_data(home)
    away_data = get_team_data(away)

    # ── 80% Skill Analysis ──
    home_power = calculate_power_score(home, away)
    away_power = calculate_power_score(away, home)

    # Head-to-head adjustment (home advantage ~5%)
    home_advantage = 5
    home_power += home_advantage
    away_power -= home_advantage * 0.3

    total_power = home_power + away_power
    home_skill_pct = round((home_power / total_power) * 100)
    away_skill_pct = 100 - home_skill_pct

    # ── 10% Luck / Fortune ──
    luck_seed = random.randint(1, 100)
    is_home_lucky = luck_seed > 50
    luck_value = random.randint(-8, 12)  # -8% ~ +12%
    home_luck = luck_value if is_home_lucky else -luck_value
    away_luck = -home_luck

    luck_msg = random.choice(LUCK_GOOD) if home_luck > 0 else random.choice(LUCK_BAD)
    if abs(home_luck) < 3:
        luck_msg = "운세는 평범합니다 😐"

    # ── 10% Weather ──
    weather = random.choice(WEATHERS)
    home_weather_pct = round(weather["effect_home"] * 10)
    away_weather_pct = round(weather["effect_away"] * 10)

    # ── Final Probability ──
    home_total = home_skill_pct * 0.8 + (home_luck + 50) * 0.1 + home_weather_pct * 0.1
    away_total = away_skill_pct * 0.8 + (away_luck + 50) * 0.1 + away_weather_pct * 0.1
    draw_total = 20  # base draw probability

    total = home_total + away_total + draw_total
    home_pct = round((home_total / total) * 100)
    away_pct = round((away_total / total) * 100)
    draw_pct = 100 - home_pct - away_pct

    # ── Predicted Score ──
    home_goals = round((home_power / 100) * random.uniform(1.5, 3.5))
    away_goals = round((away_power / 100) * random.uniform(0.5, 2.5))
    if home_goals == away_goals:
        if random.random() > 0.5:
            home_goals += 1
        else:
            away_goals += 1

    # ── Analysis Text ──
    if home_power - away_power > 15:
        verdict = f"🏆 {home}의 압도적 우세! 실력 차이가 확연합니다."
    elif home_power - away_power > 5:
        verdict = f"⭐ {home}이(가) 다소 유리합니다. 홈에서 경기하는 이점도 있습니다."
    elif abs(home_power - away_power) <= 5:
        verdict = "⚖️ 초접전이 예상됩니다! 누가 이겨도 이상하지 않습니다."
    elif away_power > home_power:
        verdict = f"💪 {away}이(가) 객관적 전력에서 앞서지만, 변수가 많습니다."
    else:
        verdict = "🔮 예측이 어려운 매치업입니다."

    return {
        "match_id": match.get("id", ""),
        "home_team": home,
        "away_team": away,
        "predicted_score": {"home": home_goals, "away": away_goals},
        "winner": home if home_goals > away_goals else (away if away_goals > home_goals else "무승부"),
        "probabilities": {
            "home_win": home_pct,
            "draw": draw_pct,
            "away_win": away_pct,
        },
        "analysis": {
            "skill_80": {
                "home_power": round(home_power),
                "away_power": round(away_power),
                "home_attack": f"⚔️ {home_data['gf_avg']:.1f} goals/match ({home_data['style']})",
                "away_attack": f"⚔️ {away_data['gf_avg']:.1f} goals/match ({away_data['style']})",
                "home_defense": f"🛡️ {home_data['ga_avg']:.1f} conceded/match",
                "away_defense": f"🛡️ {away_data['ga_avg']:.1f} conceded/match",
                "home_form": f"최근 {home_data['form']}",
                "away_form": f"최근 {away_data['form']}",
                "key_player": f"⭐ {home_data['key_player']} vs {away_data['key_player']}",
                "rank_diff": f"FIFA 랭킹: {home_data['rank']}위 vs {away_data['rank']}위",
                "home_pct": home_skill_pct,
                "away_pct": away_skill_pct,
            },
            "luck_10": {
                "message": luck_msg,
                "value": round(home_luck * 2),
                "is_lucky": home_luck > 0,
            },
            "weather_10": {
                "condition": weather["condition"],
                "temperature": weather["temp"],
                "home_advantage": weather["effect_home"],
                "away_disadvantage": weather["effect_away"],
            },
        },
        "verdict": verdict,
        "confidence": "매우 높음" if abs(home_power - away_power) > 20 else
                      "높음" if abs(home_power - away_power) > 10 else
                      "보통" if abs(home_power - away_power) > 5 else "낮음",
    }


@router.get("/predict/ai/analyze")
async def analyze_predictions():
    """Get AI predictions for all upcoming matches"""
    from routes.matches import MOCK_MATCHES
    upcoming = [m for m in MOCK_MATCHES if m["status"] != "finished"]
    return {"predictions": [calculate_prediction(m) for m in upcoming]}


@router.get("/predict/ai/{match_id}")
async def analyze_match(match_id: str):
    """Get AI prediction for a specific match"""
    from routes.matches import MOCK_MATCHES
    for m in MOCK_MATCHES:
        if m["id"] == match_id:
            return calculate_prediction(m)
    return {"error": "Match not found"}
